pythonforandroid/recipes/cffi/__init__.py

```diff
@@ -1,48 +1,26 @@
-import os
-from pythonforandroid.recipe import CompiledComponentsPythonRecipe
-
-
-class CffiRecipe(CompiledComponentsPythonRecipe):
-    """
-    Extra system dependencies: autoconf, automake and libtool.
-    """
-    name = 'cffi'
-    version = '1.15.1'
-    url = 'https://pypi.python.org/packages/source/c/cffi/cffi-{version}.tar.gz'
-
-    depends = ['setuptools', 'pycparser', 'libffi']
-
-    patches = ['disable-pkg-config.patch']
-
-    # call_hostpython_via_targetpython = False
-    install_in_hostpython = True
-
-    def get_hostrecipe_env(self, arch=None):
-        # fixes missing ffi.h on some host systems (e.g. gentoo)
-        env = super().get_hostrecipe_env(arch)
-        libffi = self.get_recipe('libffi', self.ctx)
-        includes = libffi.get_include_dirs(arch)
-        env['FFI_INC'] = ",".join(includes)
-        return env
-
-    def get_recipe_env(self, arch=None):
-        env = super().get_recipe_env(arch)
-        libffi = self.get_recipe('libffi', self.ctx)
-        includes = libffi.get_include_dirs(arch)
-        env['CFLAGS'] = ' -I'.join([env.get('CFLAGS', '')] + includes)
-        env['CFLAGS'] += ' -I{}'.format(self.ctx.python_recipe.include_root(arch.arch))
-        env['LDFLAGS'] = (env.get('CFLAGS', '') + ' -L' +
-                          self.ctx.get_libs_dir(arch.arch))
-        env['LDFLAGS'] += ' -L{}'.format(os.path.join(self.ctx.bootstrap.build_dir, 'libs', arch.arch))
-        # required for libc and libdl
-        env['LDFLAGS'] += ' -L{}'.format(arch.ndk_lib_dir_versioned)
-        env['PYTHONPATH'] = ':'.join([
-            self.ctx.get_site_packages_dir(arch),
-            env['BUILDLIB_PATH'],
-        ])
-        env['LDFLAGS'] += ' -L{}'.format(self.ctx.python_recipe.link_root(arch.arch))
-        env['LDFLAGS'] += ' -lpython{}'.format(self.ctx.python_recipe.link_version)
-        return env
+from os.path import exists, join
+from pythonforandroid.logger import info
+from pythonforandroid.recipe import PythonRecipe
+
+
+# A very hacky recipe
+# This avoids hassle of finding and correctly linking libffi
+# and just works
+
+
+class CffiRecipe(PythonRecipe):
+    version = "1.15.1"
+    hostpython_prerequisites = ["cffi=={}".format(version), "pycparser"]
+
+    def should_build(self, arch):
+        if exists(join(self.hostpython_site_dir, "cffi")):
+            info('Python package already exists in site-packages')
+            return False
+        info('cffi apparently isn\'t already in site-packages')
+        return True
```

Comment:
Can you explain to me why you added these check to almost any recipe you changed?

Do we need to apply such a check on the base `Recipe` code for a specific reason?

Reply:
setuptools, cffi: both are installed in hostpython, but `should_build` of `Recipe` checks in `python-installs` folder.
pydantic-core: 
```diff
--- if self.ctx.has_package(name, arch):
+++ if self.ctx.has_package(name.replace("-", "_"), arch):
```
(because folder name is not same as name)

Reply:
Ok, but both `cffi` and `setuptools` can also be built and installed into device `site-packages` folder, and not only for `hostpython`.

(Mostly needed for `cffi`).

As I can see, you removed the whole recipe for `cffi` and expect to have it installed for `hostpython3` only. I'm right?

What happens if the user specifically ask to have `setuptools` and `cffi` available on the target device?